$NetBSD: patch-ac,v 1.4 2014/09/11 21:27:09 dholland Exp $

- configure for pkgsrc
- always use tradcpp
- quote $(CC) bmake-style when passing to cpp
- allow putting stuff in CPP_ARGS from the pkgsrc makefile
- improve post-cpp seddery (?)

--- Makefile.orig	1999-03-29 06:09:39.000000000 +0000
+++ Makefile
@@ -44,22 +44,8 @@ DEFINES = -DSVR4
 DEFINES = 
 #endif
 
-#ifdef NeXT
-CC = cc -bsd
-CPP = $(CC) -traditional -E 
-#elifdef sun
-CC = gcc -g $(DEFINES)
-CPP = $(CC) -traditional -E
-#elifdef sgi
-CC = cc -g3 $(DEFINES)
-CPP = $(CC) -E -cckr
-#elifdef bsdi
 CC = cc
-CPP = $(CC) -traditional -E 
-#else
-CC = cc $(DEFINES)
-CPP = $(CC) -E
-#endif
+CPP = tradcpp
 
 DIRNAME = /tmp/dirname
 EXCLUDE = /tmp/exclude
@@ -82,7 +68,7 @@ CPP_DEFINES = -DHAVE_GCC -DHAVE_FIXED_IN
 #MAKE = make
 
 CPPFLAGS = $(SYSTEM) $(CPP_DEFINES)
-CPP_ARGS = -DCPP_CC="$(CC)" -DIncludes=$(INCLUDES)
+CPP_ARGS += -DCPP_CC=${CC:Q} -DIncludes=$(INCLUDES)
 
 all:	$(SRC)$(MKFILE)
 #	(cd $(SRC); $(MAKE) -f $(MKFILE) depend)
@@ -98,7 +84,7 @@ dirs:	FRC
 $(SRC)$(MKFILE): support/cpp-Makefile Makefile config.h
 	cp $(SUPPORT)cpp-Makefile $(SRC)tmp-Makefile.c
 	(cd $(SRC); $(CPP) $(CPP_ARGS) tmp-Makefile.c \
-	| sed 's@[ 	]*$$@@' | uniq >$(MKFILE))
+	| sed -e 's@[ 	]*$$@@' -e 's@^         *@	@' | uniq >$(MKFILE))
 	rm -f $(SRC)tmp-Makefile.c
 
 # Subsequent lines contain targets that are correctly handled by an
